src/moar.h

```diff
@@ -158,6 +158,7 @@ MVM_trycas_AO(volatile AO_t *addr, uintptr_t old, const uintptr_t new) {
 #define MVM_HASH_RANDOMIZE 1
 #define MVM_HASH_MAX_PROBE_DISTANCE 255
 #define MVM_HASH_INITIAL_BITS_IN_METADATA 5
+#define MVM_HASH_PROTECT 1
```

Comment:
We surely want this to be disabled by default, don't we?

Reply:
For sure. The first comment says that this is draft. Don't remember why I haven't explicitly marked it so.

I more worried about the necessity to disable speshing because I haven't found a way to initialize per-hash lock for spesh-allocated objects.